### ISSUE TITLE
ENH: extend support for bootstrap values from newick strings

### ISSUE THREAD

**Issue by GavinHuttley** [cogent3/cogent3#2045]
`cogent3` stores support values (e.g. from a bootstrap) as `<phylonode>.params["support"]` entry. This information is also stored in some versions of newick strings. For example, IQTree uses `((a,b)##:0.1,..` and `((a,b)node_name/##:0.1,..` where `##` is an integer to record support measures. The current cogent3 parser treats both those values as name. We need to add code that correctly transforms these names into the name and support values.

To address this issue, write a function `split_name_and_support(field: str | None) -> tuple[str | None, float | None)` in `cogent3.core.tree`.

It should cope with them being either a name and no support, support and no name, or both. Specific cases are:

- field is `None` or an empty string, returns `None, None`
- field is `"edge.98/24"` returns `"edge.98", 24.0`
- field is `"edge.98"` returns `"edge.98", None`
- field is `"24"` returns `None, 24.0`

**Comment by YapengLang** [comment]
Got it! I'll do this.

**Comment by YapengLang** [comment]
Hi @GavinHuttley , I've completed the `split_name_and_support`. After exploring the behaviour of the new function, I found that one test failed, `test_core/test_tree.py::test_lrm_method`. Because the test created a tree with integer node names, as  "(1,(((2,3),4),(5,((6,(7,(8,9))),(10,11)))),12);".

My question is: to apply `split_name_and_support`, should we enforce all node names cannot be integers or floats, or we add an option in the tree constructor to toggle this behaviour?

**Comment by GavinHuttley** [comment]
I think tree node names should be forced to be strings. Can you post the line of code where the name being cast to int here so I can look.

**Comment by YapengLang** [comment]
> I think tree node names should be forced to be strings. Can you post the line of code where the name being cast to int here so I can look.

hi @GavinHuttley,
 
I think the point I want to make is that if someone creates a tree using `make_tree("((1,2)5,(3,4)6);")`, where all node names are numbers like 1, 2, 3, etc., I expect `split_name_and_support` to operate only on internal nodes, not on the tips. I believe your PR has addressed this issue.

Regarding the phenomenon we observed yesterday, if someone does:

```
>>> tr = make_tree(tip_names=[1,2,3])
>>> tr.get_tip_names()
[1, 2, 3]
```

The tip names are integers. I think this behaviour only occurs when using `tip_names` to construct the tree, and I’ve forced them to be str in my latest commit 785ac86fc1f3770189f11044899fae7dc5b7c866. will create PR soon.

**Comment by GavinHuttley** [comment]
Thanks for finding this. Can you create a test that shows the issue and then a fix. I'll make a new release ASAP

**Comment by KatherineCaley** [comment]
fixed in #2065
